javascript/ql/src/semmle/javascript/ApiGraphs.qll

```diff
@@ -413,6 +417,13 @@ module API {
         not n.isRoot() and
         isUsed(n)
       } or
+      /**
+       * A TypeScript type, identified by name of the type-annotation.
+       * This API node is exclusively used by `API::Node::ofType`.
+       */
+      MkHasUnderlyingType(string moduleName, string exportName) {
+        any(TypeAnnotation n).hasQualifiedName(moduleName, exportName)
```

Comment:
We could use `any(Type t).hasUnderlyingType` in addition to the above, for cases where the value comes in through a callback parameter without a type annotation, but where the type is inferred by contextual typing.

Or will it generate too many unneeded API nodes?